extensions/cpsection/network/view.py

```diff
@@ -18,17 +18,268 @@
+        check_button.show()
+        self.pack_start(check_button, True, True, 0)
+        self.pack_start(contents_box, False, False, 0)
+
+        setting.bind(setting_key, check_button, 'active',
+                     Gio.SettingsBindFlags.DEFAULT)
+
+    def __button_toggled_cb(self, check_button, contents_box):
+        contents_box.set_visible(check_button.get_active())
+
+
+class HostPortSettingBox(SettingBox):
+    """
+    A configuration line for a combined host name and port setting.
+    """
+    def __init__(self, name, alert, setting, size_group=None):
+        SettingBox.__init__(self, name, size_group)
+        self.pack_start(alert, False, True, 0)
+        alert.hide()
+
+        host_entry = Gtk.Entry()
+        self.pack_start(host_entry, True, True, 0)
+        host_entry.show()
+
+        setting.bind('host', host_entry, 'text', Gio.SettingsBindFlags.DEFAULT)
+
+        # port number 0 means n/a
+        port_entry = NumberEntry()
+        self.pack_start(port_entry, False, False, 0)
+        port_entry.show()
+        setting.bind_with_convert('port',
+                                  port_entry,
+                                  "text",
+                                  Gio.SettingsBindFlags.GET |
+                                  Gio.SettingsBindFlags.SET |
+                                  Gio.SettingsBindFlags.NO_SENSITIVITY,
+                                  lambda x: str(x),
+                                  lambda x: int(x))
+
+
+class StringSettingBox(SettingBox):
+    """
+    A configuration line for a string setting.
+    """
+    def __init__(self, name, setting, setting_key, size_group=None,
+                 password_field=False):
+        SettingBox.__init__(self, name, size_group)
+
+        entry = Gtk.Entry()
+        self.pack_start(entry, True, True, 0)
+        entry.show()
+        if password_field:
+            entry.set_visibility(False)
+
+        setting.bind(setting_key, entry, 'text', Gio.SettingsBindFlags.DEFAULT)
+
+
+class StringSettingBox_with_convert(SettingBox):
+    """
+    A configuration line for a string setting.
+    """
+    def __init__(self, name, setting, setting_key,
+                 get_method, set_method, size_group=None,
+                 password_field=False):
+        SettingBox.__init__(self, name, size_group)
+
+        entry = Gtk.Entry()
+        self.pack_start(entry, True, True, 0)
+        entry.show()
+        if password_field:
+            entry.set_visibility(False)
+
+        setting.bind_with_convert(setting_key,
+                                  entry,
+                                  "text",
+                                  Gio.SettingsBindFlags.GET |
+                                  Gio.SettingsBindFlags.SET |
+                                  Gio.SettingsBindFlags.NO_SENSITIVITY,
+                                  get_method,
+                                  set_method)
+
```

Comment:
All of the code above here seems like helpful code that could be put in a shared library.

After this patch is merged, it could be interesting to go on a refactoring run and put this in a shared library!  We could clean up a lot of control panels using this.